src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,11 +80,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String userText =
+        Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
```

Comment:
It's possible that this will fail if the URL is has a quote in it, but I'm not sure. You may want to test.